added feature to replace dead agent

shamith2 opened the pull request:
#24 
agent.py ('move' method): added feature to replace dead agent with home background

shamith2 committed:
changes made so that UAV and UGV do not collide

skim0119 committed:
Merge branch 'Release' of https://github.com/raide-project/ctf_public into issue17

shamith2 committed:
added feature to replace dead agent 

move method: added feature to replace dead agent with home background

skim0119 commented on gym_cap/envs/agent.py:
```diff
@@ -47,6 +47,7 @@ def move(self, action, env, team_home):
             easily place the correct home tiles
         """
         if not self.isAlive:
+            env[self.x][self.y] = team_home[self.x][self.y]   # agent replaced with home background after it dies
```
If agent pass through the same spot, it does not render properly.

shamith2 committed:
update agent.py; still working

skim0119 committed:
simplify move method for debugging;

skim0119 committed:
Merge branch 'Release' of https://github.com/raide-project/ctf_public into issue17

shamith2 committed:
update agent.py; still working

shamith2 committed:
update agent.py; still working

agent.py: added 'memory' array to remember the locations an agent previously visited.

shamith2 committed:
update agent.py; still working

shamith2 committed:
update agent.py: still working

skim0119 committed:
Merge commit 'd421a6c' into issue24

skim0119 committed:
resolve conflict

skim0119 committed:
add if statement to just change the color when dead block is on the way
